modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/footprint/SidecarFootprintProvider.java

```diff
@@ -219,7 +219,7 @@ private String getFullPath(String strValue) throws IOException {
         if (!file.isAbsolute()) {
             file = new File(reference, strValue);
         }
-        return file.getCanonicalPath();
+        return file.getPath();
```

Comment:
I think that the `file.getCanonicalPath()` to `file.getPath()` change in `SidecarFootprintProvider.getFullPath` is the wrong fix because this calling code should not be passing the string representation of an HTTP(S) URL in the first place. I would prefer a guard in the calling code to prevent HTTP(S) URLs from being used until they are supported. For example, change line 117 to:
```
            if (value != null && value instanceof String
                    && !((String) value).matches("https?://.*")) {
```

Reply:
Done